Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -469,7 +482,14 @@ public final class PackageBuilder {
                 if subpath == "" || subpath == "." {
                     return packagePath
                 }
-                let path = packagePath.appending(RelativePath(subpath))
+                
+                // Make sure the target path doesn't contain '~'.
+                guard subpath.first != "~" else {
```

Comment:
I think we should create a static method in AbsolutePath to check for invalid path characters so other parts of the code can also make use of the functionality.

Reply:
How then return the reason of why the path is not supported? 
``` swift
enum PathCheckResult {
    case supported
    case unsupported(InvalidPathType)
}

extension AbsolutePath {
    static func check(path: String) -> PathCheckResult {
       ...
    }
}
```

Reply:
Since we want to throw in the end, you can make the static method throwing and encapsulate the error that way.